src/query/shorthand.ts

```diff
@@ -53,7 +56,13 @@ export function spec(specQ: SpecQuery,
     parts.push(value(specQ.mark, replace[Property.MARK]));
   }
 
-  // TODO: transform
+  if (specQ.transform) {
+    const calc = stringifyCalculate(specQ.transform.calculate);
+    const filter = stringifyFilter(specQ.transform.filter);
+
+    parts.push((calc ? `calculate:${calc}` : '') +
```

Comment:
As we promote them to top-level n shorthand.  Why don't you consider these as different parts that get pushed separately.  (Thus get  top-level `|` delimiter automatically.)